### DISCUSSION TITLE
DHT support

### DISCUSSION THREAD

**Discussion by proasnet** [HASwitchPlate/HASPone#212] [Ideas]
Hi, is possible add a DHT sensor support?
I have a four HASP in my home, and I want get to this rooms temperature and humidity sensor for HA. But, is better use only one ESP in a room, not next.
Thank you for your project.

**Comment by aderusha** [comment]
Short answer: probably not.

One would need to solve a lot of problems to make this work:

* DHT11 is huge, power hungry, and not terribly accurate.  Something like a BME280 etc might solve for this.
* NEC requires that all conductors stay in the enclosure, and we can't run wires for low voltage out of the enclosure.  So, any sensor would need to find a place to fit that is entire inside the enclosure, possibly with holes to allow air to flow in.
* That device would also need to face the room or you'd just be measuring the temperature/humidity inside the box
* The front plate is 100% consumed by the display through the opening in your work box, so you'd also need to find more space that isn't already used to place that sensor.

**Comment by proasnet** [comment]
It doesn't have to be DHT11, but for example DHT22.
In my case, I have enough space in the box, I have my own made according to the EU size, and I also have no problem having the sensor right next to the box or on its side wall, these are just construction issues.
As you have motion sensor support, in my case I used this option to connect a capacitiy touch sensor next to the display, so it would be great to be able to enable the DHT22 function and send the data to MQTT.
